File: TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/data/ToyDao.java
package com.enpassio.twowaydatabinding.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.enpassio.twowaydatabinding.data.model.ToyEntry;

import java.util.List;

@Dao
public interface ToyDao {

    @Query("SELECT * FROM toys")
    LiveData<List<ToyEntry>> getAllToys();

    @Query("SELECT * FROM toys WHERE toyId = :id")
    LiveData<ToyEntry> getChosenToy(int id);

    @Insert
    void insertToy(ToyEntry toy);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateToyInfo(ToyEntry toy);

    @Delete
    void deleteToy(ToyEntry toy);
}

File: TwoWayDatabindingKotlin/app/src/main/java/com/enpassion/twowaydatabindingkotlin/data/ToyDao.kt
package com.enpassion.twowaydatabindingkotlin.data

import androidx.lifecycle.LiveData
import androidx.room.*


@Dao
interface ToyDao {

    @get:Query("SELECT * FROM toys")
    val allToys: LiveData<List<ToyEntry>>

    @Query("SELECT * FROM toys WHERE toyId = :id")
    fun getChosenToy(id: Int): LiveData<ToyEntry>

    @Insert
    fun insertToy(toy: ToyEntry)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateToyInfo(toy: ToyEntry)

    @Delete
    fun deleteToy(toy: ToyEntry)
}